packages/flutter/lib/src/material/page.dart

```diff
@@ -20,8 +20,10 @@ import 'theme.dart';
 /// in environments with a right-to-left reading direction.)
 ///
 /// By default, when a modal route is replaced by another, the previous route
-/// remains in memory. To free all the resources when this is not necessary, set
-/// [maintainState] to false.
+/// remains in memory. To free resources when this is not necessary, set
+/// [maintainState] to false. When [maintainState] is false, it stills maintains
+/// the first, and only the first, route in memory. Use [Navigator.pushReplacement]
```

Comment:
This statement sounds misleading. When maintainState is set to false on a particular route, other routes below it may still be kept in memory (e.g. because they may have maintainState set to true). And even if this route has maintainState set to false and is not the topmost route it may be kept in memory when the route above it is not opaque. See also https://master-api.flutter.dev/flutter/widgets/OverlayEntry/maintainState.html.